06_puppy/CameronSchafer/main.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sync" //https://golang.org/pkg/sync/#Map
+)
+
+var out io.Writer = os.Stdout
+
+type Puppy struct {
+	pid    int
+	breed  string
+	colour string
+	value  string
+}
+
+type MapStore struct {
+	store map[int]Puppy
+}
+
+type SyncStore struct {
+	store sync.Map
+}
+
+type Storer interface {
+	CreatePuppy(*Puppy) int
+	ReadPuppy(int) *Puppy
+	UpdatePuppy(*Puppy)
+	DeletePuppy(int)
```

Comment:
These interface functions do not return error at all. Wonder how does function caller know something is wrong, e.g. id not found?

Reply:
Yup please fix.